src/readers/morphologyHDF5.cpp

```diff
@@ -417,12 +427,11 @@ void MorphologyHDF5::_readPerimeters(int firstSectionOffset) {
 
     try {
         HighFive::SilenceHDF5 silence;
-        HighFive::DataSet dataset = _file->getDataSet(_d_perimeters);
+        HighFive::DataSet dataset = _group.getDataSet(_d_perimeters);
 
         auto dims = dataset.getSpace().getDimensions();
         if (dims.size() != 1) {
-            throw MorphioError("Reading morphology file '" + _file->getName() +
```

Comment:
What about setting the filename as a class attribute and setting it to the empty string (or something more meaningful like "HDF5 Group") when reading from a group ? This way we could keep the old error message.
I think it is an important feature to know on which file it failed when you are iterating over a very large number of morphologies and it suddenly fails.

Reply:
Makes sense. I will work on it.